### PR TITLE
Minor OTel Agent Fixes

### PR DESCRIPTION
**Pull request by dineshg13**
### What does this PR do?

The PR fixes the following in the OTel Agent 

- Adds DRY run to the collector, to ensure OTel Agent fails when we run with unknown components or bad collector config. 
- Adds ability to fetch hostname from core agent using grpc. 
- Configures the OTel collector to use Agent logger.

### CODE REVIEW

File: cmd/otel-agent/subcommands/run/command.go [hunk 1]
```diff
@@ -76,6 +79,22 @@ func (o *orchestratorinterfaceimpl) Reset() {
 	o.f = nil
 }
 
+type remotehostimpl struct {
```
Comment (truthbk) [inline cmd/otel-agent/subcommands/run/command.go:82]: While this is probably OK; at least for now this does seem like something that should be an actual alternative implementation of the current hostname.Component, or the hostname.Component should've been parametrized such that remote fetching could be possible.

This is logic that is used in several other agents, and so having it exposed as code that could be easily reusable by other agents would make sense to me.

Not a blocker, but maybe worth a TODO comment at least.

**Review by truthbk — COMMENTED** [review]
Looks good for the most; just a minor comment.

File: cmd/otel-agent/subcommands/run/command.go [hunk 2]
```diff
@@ -109,10 +128,14 @@ func runOTelAgentCommand(_ context.Context, params *subcommands.GlobalParams, op
...
 		}),
+		fx.Provide(func() (serializerexporter.SourceProviderFunc, error) {
+			rh := &remotehostimpl{}
+			return rh.Get, nil
+		}),
 
 		fx.Supply(optional.NewNoneOption[secrets.Component]()),
 		fx.Provide(func(c config.Component) corelogimpl.Params {
-			return corelogimpl.ForOneShot(params.LoggerName, c.GetString("log_level"), true)
+			return corelogimpl.ForDaemon(params.LoggerName, "log_file", pkgconfigsetup.DefaultOTelAgentLogFile)
```
[the remark this answers is not in the document]
Comment (dineshg13) [reply cmd/otel-agent/subcommands/run/command.go:138]: Can we do it in the PR that adds trace agent component ?

File: cmd/otel-agent/subcommands/run/command.go (hunk 1 above)
Comment (dineshg13) [reply cmd/otel-agent/subcommands/run/command.go:85]: Added TODO

File: cmd/otel-agent/subcommands/run/command.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (songy23) [reply cmd/otel-agent/subcommands/run/command.go:140]: Yes. I can send a separate PR to add trace agent comp to fx first

**Review by songy23 — APPROVED** [review]

**Review by truthbk — APPROVED** [review]

**Pull request merged**